allspice/allspice.py

```diff
@@ -152,8 +152,37 @@ def requests_delete(self, endpoint: str):
             self.logger.error(message)
             raise Exception(message)
 
-    def requests_post(self, endpoint: str, data: dict):
-        request = self.requests.post(self.__get_url(endpoint), headers=self.headers, data=json.dumps(data))
+    def requests_post(
+            self,
+            endpoint: str,
+            data: Optional[dict] = None,
+            params: Optional[dict] = None,
+            files: Optional[dict] = None,
+    ):
+        """
+        Make a POST call to the endpoint.
+
+        :param endpoint: The path to the endpoint
+        :param data: A dictionary for JSON data
+        :param params: A dictionary of query params
+        :param files: A dictionary of files, see requests.post. Using both files and data
+                      can lead to unexpected results!
+        :return: The JSON response parsed as a dict
+        """
+
+        args = {
+            "headers": self.headers.copy(),
```

Comment:
Why shallow copy instead of passing the reference?

Reply:
We `pop` a key from this dict in line 181, so if we re-used the reference that header would be missing from other requests which should have it. A shallow copy is fine here because it's just string keys to string values, and we don't change any of the values anyway.